service/src/test/java/com/codeforcommunity/dataaccess/AuthDatabaseOperationsTest.java

```diff
@@ -1,171 +1,155 @@
+        myUser.setPassHash(Passwords.createHash("password"));
+        myUser.setId(1);
+        myUser.setPrivilegeLevel(PrivilegeLevel.STANDARD);
+        myJooqMock.addReturn("SELECT", myUser);
+
+        assertFalse(myAuthDatabaseOperations.isValidLogin(myEmail, "wrongPassword"));
+    }
+
+    // returns true for correct login
+    @Test
+    public void testIsValidLogin2() {
+        String myEmail = "kimin@example.com";
+
+        // one user in DB
+        UsersRecord myUser = myJooqMock.getContext().newRecord(Tables.USERS);
+        myUser.setUsername("kiminusername");
+        myUser.setEmail(myEmail);
+        myUser.setPassHash(Passwords.createHash("password"));
+        myUser.setId(1);
+        myUser.setPrivilegeLevel(PrivilegeLevel.STANDARD);
+        myJooqMock.addReturn("SELECT", myUser);
+
+        assertTrue(myAuthDatabaseOperations.isValidLogin(myEmail, "password"));
+    }
+
+    // creating a new user fails when the email is already in use
+    @Test
+    public void testCreateNewUser1() {
+        String myEmail = "kimin@example.com";
+
+        // one user in DB
+        UsersRecord myUser = myJooqMock.getContext().newRecord(Tables.USERS);
+        myUser.setUsername("kiminusername");
+        myUser.setEmail(myEmail);
+        myUser.setPassHash(Passwords.createHash("password"));
+        myUser.setId(1);
+        myUser.setPrivilegeLevel(PrivilegeLevel.STANDARD);
+        myJooqMock.addReturn("SELECT", myUser);
+
+        try {
+            myAuthDatabaseOperations.createNewUser("diffusername", myEmail, "password", "Kimin", "Lee");
+            fail();
+        } catch (EmailAlreadyInUseException e) {
+            assertEquals(e.getEmail(), "kimin@example.com");
+        }
+    }
+
+    // creating a new user fails when the username is already in use
+    @Test
+    public void testCreateNewUser2() {
+        String myEmail = "kimin@example.com";
+        String myUsername = "kiminusername";
+
+        // one user in DB
+        UsersRecord myUser = myJooqMock.getContext().newRecord(Tables.USERS);
+        myUser.setUsername(myUsername);
+        myUser.setEmail(myEmail);
+        myUser.setPassHash(Passwords.createHash("password"));
+        myUser.setId(1);
+        myUser.setPrivilegeLevel(PrivilegeLevel.STANDARD);
+        myJooqMock.addReturn("SELECT", myUser);
+
+        try {
+            myAuthDatabaseOperations.createNewUser(myUsername, "diff@example.com", "password", "Kimin", "Lee");
+            fail();
+        } catch (EmailAlreadyInUseException e) {
+            assertEquals(e.getEmail(), "diff@example.com");
+        }
+    }
+
+    // creating a new user succeeds when the email and username isn't already in use
+    @Test
+    public void testCreateNewUser3() {
+        // no users in DB
+        myJooqMock.addReturn("SELECT", new ArrayList<UsersRecord>());
+
+        myAuthDatabaseOperations.createNewUser("connerusername", "conner@example.com", "password", "Conner", "Nilsen");
```

Comment:
Is there anything you can test about this?

Reply:
Checked the user data and times called for "SELECT"